targets/spidermonkey/templates/layout_head.c

```diff
@@ -28,8 +28,13 @@ static bool dummy_constructor(JSContext *cx, uint32_t argc, jsval *vp) {
         typeClass = typeMapIter->second;
         CCASSERT(typeClass, "The value is null.");
 
+\#if (COCOS2D_VERSION >= 0x00031000)
```

Comment:
this one looks hard to add a compatibility layer... I'll see if I can do something for it.